File: EJEMPLOS PATRONES/Python/State.py
import abc


class Player:
    def __init__(self, state):
        self.__state = state

    def request(self):
        self._state.handle()


class State(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def executeCommand(self):
        pass


class DeadState(State):
    def executeCommand(self):
        print("this player is dead")


class HurtState(State):
    def executeCommand(self):
        print("this player is hurt, do command slowly")


class HealthyState(State):
    def executeCommand(self):
        print("this player is Healthy, do command so fast")

File: Parte1/EJEMPLOS PATRONES/PHP/state.php
<?php

class Player {

    private $state = null;

    public function __construct($state) {
        $this->state = $state;
    }

    public function getState() {
        return $this->state;
    }

}

interface State {

    public function executeCommand();
}

class DeadState implements State {

    public function executeCommand() {
        echo "this player is dead";
    }

}

class HurtState implements State {

    public function executeCommand() {
        echo "this player is hurt, do command slowly";
    }

}

class HealtyState implements State {

    public function executeCommand() {
        echo "this player is Healthy, do command so fast";
    }

}
